From 558ed34d4e99c8b42c0fb07106ec24ff8579d880 Mon Sep 17 00:00:00 2001
From: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Date: Mon, 12 Jun 2023 11:40:25 +0200
Subject: [PATCH 306/954] feat(fdts): rename STM32MP257F-EV into
 STM32MP257F-EV1

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 41c7419fc425c760e507d23d7d977113b094cfab feat(fdts): rename STM32MP257F-EV into STM32MP257F-EV1]
Change-Id: Ic28626f778ba4ee4799c3b8a7d5fe58d3bcc0f6b
Signed-off-by: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/312160
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 docs/plat/st/stm32mp2.rst                     | 45 +++++++++++++++++++
 ... stm32mp257f-ev1-ca35tdcid-fw-config.dtsi} |  0
 ...tsi => stm32mp257f-ev1-ca35tdcid-rcc.dtsi} |  0
 fdts/stm32mp257f-ev1-fw-config.dts            |  7 +++
 fdts/stm32mp257f-ev1.dts                      |  2 +-
 5 files changed, 53 insertions(+), 1 deletion(-)
 rename fdts/{stm32mp257f-ev-ca35tdcid-fw-config.dtsi => stm32mp257f-ev1-ca35tdcid-fw-config.dtsi} (100%)
 rename fdts/{stm32mp257f-ev-ca35tdcid-rcc.dtsi => stm32mp257f-ev1-ca35tdcid-rcc.dtsi} (100%)
 create mode 100644 fdts/stm32mp257f-ev1-fw-config.dts

diff --git a/docs/plat/st/stm32mp2.rst b/docs/plat/st/stm32mp2.rst
index f10de1d09a..e914832c1b 100644
--- a/docs/plat/st/stm32mp2.rst
+++ b/docs/plat/st/stm32mp2.rst
@@ -130,6 +130,51 @@ ___
         BL32_EXTRA1=<optee_directory>/tee-pager_v2.bin
         fip
 
+Trusted Boot Board
+__________________
+
+.. code:: shell
+
+    tools/cert_create/cert_create -n --rot-key build/stm32mp2/release/rot_key.pem \
+        --tfw-nvctr 0 \
+        --ntfw-nvctr 0 \
+        --key-alg ecdsa --hash-alg sha256 \
+        --trusted-key-cert build/stm32mp2/release/trusted_key.crt \
+        --tos-fw <optee_directory>/tee-header_v2.bin \
+        --tos-fw-extra1 <optee_directory>/tee-pager_v2.bin \
+        --tos-fw-cert build/stm32mp2/release/tos_fw_content.crt \
+        --tos-fw-key-cert build/stm32mp2/release/tos_fw_key.crt \
+        --nt-fw <u-boot_directory>/u-boot-nodtb.bin \
+        --nt-fw-cert build/stm32mp2/release/nt_fw_content.crt \
+        --nt-fw-key-cert build/stm32mp2/release/nt_fw_key.crt \
+        --hw-config <u-boot_directory>/u-boot.dtb \
+        --fw-config build/stm32mp2/release/fdts/stm32mp257f-ev1-fw-config.dtb \
+        --stm32mp-cfg-cert build/stm32mp2/release/stm32mp_cfg_cert.crt \
+        --soc-fw-config build/stm32mp2/release/fdts/stm32mp257f-ev1-bl31.dtb \
+        --soc-fw build/stm32mp2/release/bl31.bin \
+        --soc-fw-cert build/stm32mp2/release/soc_fw_content.crt \
+        --soc-fw-key-cert build/stm32mp2/release/soc_fw_key.crt \
+        --ddr-fw drivers/st/ddr/phy/firmware/bin/ddr4_pmu_train.bin
+
+    tools/fiptool/fiptool create \
+        --tos-fw <optee_directory>/tee-header_v2.bin \
+        --tos-fw-extra1 <optee_directory>/tee-pager_v2.bin \
+        --nt-fw <u-boot_directory>/u-boot-nodtb.bin \
+        --hw-config <u-boot_directory>/u-boot.dtb \
+        --fw-config build/stm32mp2/release/fdts/stm32mp257f-ev1-fw-config.dtb \
+        --soc-fw-config build/stm32mp2/release/fdts/stm32mp257f-ev1-bl31.dtb \
+        --tos-fw-cert build/stm32mp2/release/tos_fw_content.crt \
+        --tos-fw-key-cert build/stm32mp2/release/tos_fw_key.crt \
+        --nt-fw-cert build/stm32mp2/release/nt_fw_content.crt \
+        --nt-fw-key-cert build/stm32mp2/release/nt_fw_key.crt \
+        --stm32mp-cfg-cert build/stm32mp2/release/stm32mp_cfg_cert.crt \
+        --ddr-fw drivers/st/ddr/phy/firmware/bin/ddr4_pmu_train.bin \
+        --trusted-key-cert build/stm32mp2/release/trusted_key.crt \
+        --soc-fw-cert build/stm32mp2/release/soc_fw_content.crt \
+        --soc-fw-key-cert build/stm32mp2/release/soc_fw_key.crt \
+        --soc-fw build/stm32mp2/release/bl31.bin \
+        build/stm32mp2/release/fip.bin
+
 .. _Github STM32 MPUs: https://github.com/STMicroelectronics/arm-trusted-firmware/tree/HEAD/docs/plat/st/stm32mpus.rst
 
 *Copyright (c) 2023, STMicroelectronics - All Rights Reserved*
diff --git a/fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi b/fdts/stm32mp257f-ev1-ca35tdcid-fw-config.dtsi
similarity index 100%
rename from fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi
rename to fdts/stm32mp257f-ev1-ca35tdcid-fw-config.dtsi
diff --git a/fdts/stm32mp257f-ev-ca35tdcid-rcc.dtsi b/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
similarity index 100%
rename from fdts/stm32mp257f-ev-ca35tdcid-rcc.dtsi
rename to fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
diff --git a/fdts/stm32mp257f-ev1-fw-config.dts b/fdts/stm32mp257f-ev1-fw-config.dts
new file mode 100644
index 0000000000..fdc9dcfde9
--- /dev/null
+++ b/fdts/stm32mp257f-ev1-fw-config.dts
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (c) 2020-2023, STMicroelectronics - All Rights Reserved
+ */
+
+#include "stm32mp25-fw-config.dtsi"
+#include "stm32mp257f-ev1-ca35tdcid-fw-config.dtsi"
diff --git a/fdts/stm32mp257f-ev1.dts b/fdts/stm32mp257f-ev1.dts
index d29e2ca3eb..2386ef4bd5 100644
--- a/fdts/stm32mp257f-ev1.dts
+++ b/fdts/stm32mp257f-ev1.dts
@@ -9,7 +9,7 @@
 #include <dt-bindings/clock/stm32mp25-clksrc.h>
 #include "stm32mp257.dtsi"
 #include "stm32mp25xf.dtsi"
-#include "stm32mp257f-ev-ca35tdcid-rcc.dtsi"
+#include "stm32mp257f-ev1-ca35tdcid-rcc.dtsi"
 #include "stm32mp25-pinctrl.dtsi"
 #include "stm32mp25xxai-pinctrl.dtsi"
 
-- 
2.53.0

